### PR TITLE
user exit: include settings parameters in change_tadir()

### PR DESCRIPTION
**Pull request by andrecarrilho**
Including in the method change_tadir of the user exits, the 3 parameters from local settings :

1. Ignore subpackages
2. Only Local Objects
3. Only Serialize Main Language

So that they are available for the implementation of the exit.

image

### CODE REVIEW

**Comment by larshp** [comment]
@andrecarrilho please run the pretty printer for the intf code

**Comment by mbtools** [comment]
we should avoid passing objects in exits because you can't use the same code for standalone and developer version. 

instead of `io_dot` use `io_dot->get_data()`

File: src/exits/zif_abapgit_exit.intf.abap [hunk 1]
```diff
@@ -71,7 +71,7 @@ INTERFACE zif_abapgit_exit
     IMPORTING
       !iv_package TYPE devclass
       !ii_log TYPE REF TO zif_abapgit_log
-      !io_dot TYPE REF TO zcl_abapgit_dot_abapgit
+      !is_dot_data TYPE zif_abapgit_dot_abapgit=>ty_dot_abapgit
```
Comment (mbtools) [inline src/exits/zif_abapgit_exit.intf.abap:74]: let's call it `is_dot_abapgit` (like elsewhere in abapGit)

**Review by mbtools — CHANGES_REQUESTED** [review]

**Comment by mbtools** [comment]
in general, it's best to discuss new or changes to exits in an issue before coding 😉

**Comment by larshp** [comment]
> in general, it's best to discuss new or changes to exits in an issue before coding 😉

I spoke with Andre, this change is just passing 3 values, ie. 6 lines of real change => not worth opening an issue

**Comment by andrecarrilho** [comment]
> in general, it's best to discuss new or changes to exits in an issue before coding 😉

As @larshp said, we talked about it before, that's why I coded it immediately. In any case, I'll be aware of it for future reference.
And thank you also for the pointers so far on this change!

**Comment by mbtools** [comment]
no problem. just needs pretty print of `zif_abapgit_exit` 👍

File: src/exits/zif_abapgit_exit.intf.xml [hunk 2]
```diff
@@ -10,6 +10,14 @@
...
     <UNICODE>X</UNICODE>
    </VSEOINTERF>
+   <DESCRIPTIONS_SUB>
+    <SEOSUBCOTX>
+     <CMPNAME>CHANGE_TADIR</CMPNAME>
+     <SCONAME>IO_DOT</SCONAME>
+     <LANGU>E</LANGU>
+     <DESCRIPT>abapGit - .abapgit.xml</DESCRIPT>
+    </SEOSUBCOTX>
+   </DESCRIPTIONS_SUB>
```
Comment (larshp) [inline src/exits/zif_abapgit_exit.intf.xml:20]: delete this description

File: src/exits/zif_abapgit_exit.intf.abap [hunk 3]
```diff
@@ -1,177 +1,156 @@
...
+      !iv_ignore_subpackages TYPE abap_bool DEFAULT abap_false
+      !iv_only_local_objects TYPE abap_bool DEFAULT abap_false
     CHANGING
-      !ct_tadir   TYPE zif_abapgit_definitions=>ty_tadir_tt.
-
+      !ct_tadir TYPE zif_abapgit_definitions=>ty_tadir_tt .
   METHODS create_http_client
     IMPORTING
-      !iv_url          TYPE string
+      !iv_url TYPE string
```
Comment (larshp) [inline src/exits/zif_abapgit_exit.intf.abap:81]: pretty print the file with indentation? will remove this change to the file

Comment (andrecarrilho) [reply src/exits/zif_abapgit_exit.intf.abap]: indentation restored

**Review by mbtools — APPROVED** [review]
lgtm 👍

**Pull request merged**